powerlevel9k.zsh-theme

```diff
@@ -547,6 +547,26 @@ prompt_history() {
   "$1_prompt_segment" "$0" "$2" "244" "$DEFAULT_COLOR" '%h'
 }
 
+# Detection for virtualization (systemd based systems only)
+prompt_detect_virt() {
+  if ! command -v systemd-detect-virt;then
+    return
+  fi
+  local virt=$(systemd-detect-virt)
+  local color="yellow"
+  if [[ "$virt" == "none" ]]; then
+    if [[ "$(ls -di / | grep -o 2)" != "2" ]]; then
+      virt="chroot"
+      "$1_prompt_segment" "$0" "$2" "$color" "$virt" ""
```

Comment:
Yes, but there you omitted the argument for the icon (a.k.a. visual identifier, as I call it) which is optional. In the `detect-virt` segment, you omit the argument for foreground color which is a mandatory one.
As arguments of a function are positional, omitting one is always the last one (except you pass an empty parameter to the function; in ZSH terms this is an empty string, in other programming languages this is usually `null`). So passing 5 arguments omits the icon, but you pass an empty string. On the other hand you pass `$virt` as foreground color which seems wrong to me, as this should be passed one position later as the content of the segment.